src/features/resources/StepComponent.tsx

```diff
@@ -0,0 +1,462 @@
+
+  useEffect(() => {
+    // Find the options for the current step in optionsData
+    const stepData: StepOptions | undefined = optionsData.find(
+      (item) => item.step === currentStep?.stepNumber
+    );
+
+    if (stepData) {
+      if (Array.isArray(stepData.options)) {
+        setOptions(stepData.options);
+      } else if (
+        currentStep?.stepNumber < 4 &&
+        !Array.isArray(stepData.options)
+      ) {
+        const previousSelection = breadcrumbs[breadcrumbs.length - 1];
+
+        if (stepData.options) {
+          setOptions(stepData.options[previousSelection] || []);
+        }
+      } else if (
+        currentStep?.stepNumber === 5 &&
+        typeof stepData.options === 'object'
+      ) {
+        const previousSelection = breadcrumbs[breadcrumbs.length - 1];
+
+        setOptions(
+          (stepData.options[previousSelection] as unknown as string) ||
+            'No table available'
+        );
+      }
+    }
+  }, [currentStep, breadcrumbs]);
+
+  const handleOptionChange = (name: string) => {
+    setSelectedOption(name);
+  };
+
+  const handleNextClick = () => {
+    if (selectedOption) {
+      handleNext(selectedOption);
+      setSelectedOption('');
+    }
+  };
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const currentResourse = localStorage.getItem('selectedResource');
+
+      if (currentResourse) {
+        if(currentStep?.stepNumber !== 1){
+          handleBack();
+        } else {
+          handleNext(currentResourse);
+          setSelectedOption('');
+        }
+      }
+
+      localStorage.removeItem('selectedResource');
+    }
+  }, []);
+
+  return (
+    <div>
+      <Resourceswrapper>
+        <AppContentWrapper>
+          <ResourcesContainer>
+            <Box>
+              <SectionHeading
+                text="Start Practicing"
+                align="center"
+                showLeftLine={true}
+                color="#DA9694"
+                textWidth="150px"
+                gradientType="second"
+              />
+              <ResourcesHeading variant="h2">
+                {currentStep?.stepName}
+              </ResourcesHeading>
+              <Box
+                sx={{
+                  display: 'flex',
+                  alignItems: 'center',
+                  justifyContent: 'center',
+                }}
+              ></Box>
+              <Box
+                sx={{
+                  mt: '30px',
+                  display: { xs: 'none', sm: 'flex' },
+                  alignItems: 'center',
+                }}
+              >
+                <Breadcrumbs
+                  aria-label="breadcrumb"
+                  separator={
+                    <Image
+                      src={'/icons/right-arrow.svg'}
+                      alt={'icon'}
+                      width={9}
+                      height={9}
+                      style={{ transform: 'translateY(20%)' }}
```

Comment:
\These should be in a layout component and have the specific pages rendered as separate components